pkg/operator/sync.go

```diff
@@ -212,92 +222,64 @@ func (optr *Operator) syncMutatingWebhook() error {
 	return nil
 }
 
-func (optr *Operator) waitForDeploymentRollout(resource *appsv1.Deployment, pollInterval, rolloutTimeout time.Duration) error {
-	var lastError error
-	err := wait.Poll(pollInterval, rolloutTimeout, func() (bool, error) {
-		d, err := optr.deployLister.Deployments(resource.Namespace).Get(resource.Name)
-		if apierrors.IsNotFound(err) {
-			lastError = fmt.Errorf("deployment %s is not found", resource.Name)
-			klog.Error(lastError)
-			return false, nil
-		}
-		if err != nil {
-			// Do not return error here, as we could be updating the API Server itself, in which case we
-			// want to continue waiting.
-			lastError = fmt.Errorf("getting Deployment %s during rollout: %v", resource.Name, err)
-			klog.Error(lastError)
-			return false, nil
-		}
+func (optr *Operator) checkDeploymentRolloutStatus(resource *appsv1.Deployment, requeuePeriod time.Duration) (bool, time.Duration, error) {
```

Comment:
Do we really need to pass `requeuePeriod` there?

Looking on the code, i would like to reduce a number of arguments. The only place where this value using  is a line 257, otherwise it is just a constant which we plumbing through the logic without any actual usage.

Maybe we could handle this case another way? Like, return typed error and handle `requiePeriod` above in caller function. WDYT?

Reply:
No we don't, and I meant to change this so thanks for pointing it out. The old code passed in the timings so they could be adjusted in the test, this isn't needed anymore